### ISSUE TITLE
Auth component does not work in dev mode (Live)

### ISSUE THREAD

**Issue by tencorvids** [anomalyco/sst#5034]
# Auth component not working with `sst dev`

## Issue
When running an `sst.aws.Auth` component in dev mode, the API returns "sst dev is not running". The auth handler works fine when `dev: false` is set, but fails without it.

## Reproduction
I have a basic email/password auth setup using the following handler:

```ts
import { AuthHandler } from "sst/auth";
import { createSessionBuilder } from "sst/auth";
import { Hono } from "hono";
import crypto from "crypto";
import type { Adapter } from "sst/auth/adapter";
import { handle } from "hono/aws-lambda";

const users = new Map<
  string,
  { id: string; email: string; password: string }
>();

function hashPassword(password: string): string {
  return crypto.createHash("sha256").update(password).digest("hex");
}

const randomFourDigits = (): string =>
  Math.floor(1000 + Math.random() * 9000).toString();

const EmailPasswordAdapter = (): Adapter<{ email: string }> => {
  return (router: Hono) => {
    router.post("/auth", async (c) => {
      const body = await c.req.json();
      const { action, email, password } = body;

      if (action === "signup") {
        if (users.has(email)) {
          return c.json({ error: "User already exists" }, 400);
        }
        const hashedPassword = hashPassword(password);
        const id = randomFourDigits();
        users.set(email, { id, email, password: hashedPassword });
        return c.json({ claims: { id, email } });
      }

      if (action === "signin") {
        const user = users.get(email);
        if (!user) {
          return c.json({ error: "User not found" }, 404);
        }
        const hashedPassword = hashPassword(password);
        if (hashedPassword !== user.password) {
          return c.json({ error: "Invalid password" }, 401);
        }
        return c.json({ claims: { id: user.id, email } });
      }

      return c.json({ error: "Invalid action" }, 400);
    });
  };
};

export const session = createSessionBuilder<{
  user: {
    email: string;
  };
}>();

const auth = AuthHandler({
  providers: {
    emailPassword: EmailPasswordAdapter(),
  },
  callbacks: {
    auth: {
      async allowClient() {
        return true;
      },
      async success(ctx, input) {
        return ctx.session({
          type: "user",
          properties: {
            email: input.email,
          },
        });
      },
    },
  },
});

export const handler = handle(auth);
```

### Not Working Config
```ts
export default $config({
  app(input) {
    return {
      name: "sst-auth",
      removal: input?.stage === "production" ? "retain" : "remove",
      home: "aws",
    };
  },
  async run() {
    const auth = new sst.aws.Auth("Auth", {
      authenticator: {
        handler: "./src/auth.handler",
        // dev: false,
        url: true,
      },
    });
    return {
      auth: auth.authenticator.url,
    };
  },
});
```

### Working Config
```ts
export default $config({
  app(input) {
    return {
      name: "sst-auth",
      removal: input?.stage === "production" ? "retain" : "remove",
      home: "aws",
    };
  },
  async run() {
    const auth = new sst.aws.Auth("Auth", {
      authenticator: {
        handler: "./src/auth.handler",
        dev: false,  // This fixes it
        url: true,
      },
    });
    return {
      auth: auth.authenticator.url,
    };
  },
});
```

## Expected Behavior
The auth component should work in dev mode without requiring `dev: false`.

## Actual Behavior
API returns "sst dev is not running" when `dev: false` is not set.

**Comment by rupertlssmith** [comment]
Same issue but all I am trying to do is run a very simple function that returns a 200.

```
const httpApi = new sst.aws.ApiGatewayV2("MyApi");
        httpApi.route("GET /", {
            handler: "packages/functions/src/connect.main",
            dev: false
        });
```

If I do not set dev: false, it responds with {"body":"sst dev is not running"}

**Comment by rupertlssmith** [comment]
Problem is actually version 3.3.x, roll back to 3.2.76 will work around it. 3.3.x introduced a new appsync bridge and that seems likely where the problem lies.

**Comment by thdxr** [comment]
potential fix in v3.3.11

**Comment by tencorvids** [comment]
Updating to `v3.3.11` seemed to fix the issue, thanks!

**Comment by john-royal** [comment]
I had a similar issue with Live Lambda Dev not working on 3.3.x, probably for the same reason. Was about to file my own issue but I tried again with 3.3.11 and that fixed it — thank you Dax!

**Comment by pbalenzuela** [comment]
The same happened with `APIGatewayProxyHandlerV2` when using `3.3.3` but it is working now with `3.3.21`

**Comment by solomonduncan** [comment]
@pbalenzuela @thdxr I'm on `"sst": "3.3.27"` but seem to be experiencing the same issue, has this potentially broken again?

```typescript
const nsgApiGateway = new sst.aws.ApiGatewayV2("nsgApiGateway");
// ...
// working
nsgApiGateway.route("GET /hello", {
  handler: "packages/nsg-functions/src/hello-handler.handler",
  dev: false,
});

// not working
nsgApiGateway.route("GET /hello", {
  handler: "packages/nsg-functions/src/hello-handler.handler",
});
```

**Comment by solomonduncan** [comment]
Just to save anyone else the embarrassment of the day I spent dealing with trying to figure out this issue:

Make sure your shell is using the correct Node version (`node -v`).

**Comment by heroly** [comment]
Finally got it working on the following versions:
sst 3.3.21
node v22.12.0

My issue was I was running an older version of node 16.20.0. Thanks @solomonduncan  for the tip!

**Comment by jesus997** [comment]
Looks like this broke again, I am using node v22.13.0 and sst 3.5.8 and with ApiGatewayV2 when visiting the resulting url I get the error “sst dev is not running (worker: 87b8d2a7557b4d71a027dba0ab37fba7)”. This is the exact code I have:

sst.config.ts
```
export default $config({
  app(input) {
    return {
      name: "test",
      removal: input?.stage === "production" ? "retain" : "remove",
      protect: ["production"].includes(input?.stage),
      home: "aws",
      providers: {
        aws: {
          profile: input?.stage === "production" ? "prod" : "default",
          region: "us-west-2"
        }
      }
    };
  },
  async run() {
    await import("./infra/frontend")
    const { api } = await import("./infra/api");

    return {
      apiUrl: api.url
    };
  },
});
```

infra/api.ts
```
const api = new sst.aws.ApiGatewayV2("testAPI", {
  transform: {
    route: {
      handler: (args, _) => {
        args.memory ??= "512 MB"
        args.runtime ??= "python3.9"
        args.architecture ??= "arm64"
      }
    },
  },
});

api.route("ANY /{proxy+}", "packages/api/src/main.handler");
api.route("GET /docs", "packages/api/src/main.handler");
api.route("GET /openapi.json", "packages/api/src/main.handler");

export { api };
```

infra/frontend.ts
```
const EXECUTION_ROLE = "...";
const TASK_ROLE = "...";

const defaultVpc = new sst.aws.Vpc("testVPC");

const cluster = new sst.aws.Cluster("testFrontend", { vpc: defaultVpc })

const website = cluster.addService("website", {
  dev: {
    command: "npm run dev",
    directory: "packages/frontend/website"
  },
  image: {
    context: './packages/frontend',
    dockerfile: './devops/frontend/website/Dockerfile'
  },
  loadBalancer: {
    ports: [{ listen: "80/http", forward: "4321/http" }],
  },
  executionRole: EXECUTION_ROLE,
  taskRole: TASK_ROLE
});

export { cluster, website };
```

**Comment by jayair** [comment]
@jesus997 let's create a new issue? We might also need some logs for when this happens.

**Comment by alexfernandez803** [comment]
> [@jesus997](https://github.com/jesus997) let's create a new issue? We might also need some logs for when this happens.

it is also happening to me,  sst dev is not running (worker: b9789381faee488892a805eac89xxxxx)

SST 3.5.5
node version v22.12.0

**Comment by jayair** [comment]
Can you try the latest? We pushed some fixes for this.

**Comment by imohitarora** [comment]
SST - ^3.6.18
Node - v22.12.0

I am following [docs](https://openauth.js.org/docs/start/sst/) here. 

for some reason it is not resolving dependencies for issuer, in hono index.ts


`|  Build Error auth/index.handler
|  ↳ Could not resolve "hono/aws-lambda" auth/index.ts:1:23
|  ↳ Could not resolve "@openauthjs/openauth" auth/index.ts:2:23
|  ↳ Could not resolve "@openauthjs/openauth/ui/code" auth/index.ts:3:23
|  ↳ Could not resolve "@openauthjs/openauth/provider/code" auth/index.ts:4:29
|  ↳ Could not resolve "@openauthjs/openauth/storage/memory" auth/index.ts:5:30
|  ↳ Could not resolve "valibot" auth/subjects.ts:1:31
|  ↳ Could not resolve "@openauthjs/openauth/subject" auth/subjects.ts:2:31`

**Comment by vanajmoorthy** [comment]
I'm also getting the same issue when trying to set up a step function and run a lambda throught it. 

SST - ^3.6.35
Node - v22.10.0
Device - M1 MacBook Pro, Sonoma 14.6.1

I get this error as the lambda output.

```
{
  "body": "sst dev is not running (worker: 38c77d8fe93e40f39c5c78e686bfd67d)",
  "statusCode": "500"
}
```

I tried downgrading my SST version to 3.2.73 and 3.3.11 based on https://github.com/sst/sst/issues/4125 but I get the same error. I've also tried removing all my resources and deleting the `.sst` and `node_modules` folders and reinstalling but I continually get the same error. 

My sst.log also shows this if that's helpful

`time=2025-01-30T13:00:41.902Z level=INFO msg="INFO connection timeout service=appsync.connection"`
